src/main/typescript/wcardinal/ui/d-slider-horizontal.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2019 Toshiba Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Point } from "pixi.js";
+import { DSlider, DSliderOptions, DThemeSlider } from "./d-slider";
+import { DSliderTrack } from "./d-slider-track";
+import { DSliderTrackHorizontal } from "./d-slider-track-horizontal";
+
+export interface DSliderHorizontalOptions<THEME extends DThemeSliderHorizontal> extends DSliderOptions<THEME> {
+
+}
+
+export interface DThemeSliderHorizontal extends DThemeSlider {
+
+}
+
+// this is space beetween min, range and max elements
+const HORIZONTAL_PIXEL_BALANCE = 2;
+
+export class DSliderHorizontal<
+	THEME extends DThemeSliderHorizontal = DThemeSliderHorizontal,
+	OPTIONS extends DSliderHorizontalOptions<THEME> = DSliderHorizontalOptions<THEME>
+> extends DSlider<THEME, OPTIONS> {
+
+	protected newTrack( options?: OPTIONS ): DSliderTrack {
+		return new DSliderTrackHorizontal( options && options.track );
+	}
+
+	protected newTrackSelected( options?: OPTIONS ): DSliderTrack {
+		return new DSliderTrackHorizontal( options && options.track );
+	}
+
+	protected assertSize(): void {
+		this._track.setWidth( this.width );
+		this._trackSelected.setHeight( this._track.height );
+		this._trackSelected.setWidth( this._track.width );
```

Comment:
The `trackSelected`'s width is updated in the methods `assertSize`, `updateCoordinates` and `updateChildren`.
Are you sure that they are all required?